server/src/main/java/org/elasticsearch/common/util/concurrent/QueueResizingEsThreadPoolExecutor.java

```diff
@@ -151,8 +151,12 @@ protected void afterExecute(Runnable r, Throwable t) {
         final long totalNanos = totalTaskNanos.addAndGet(taskNanos);
 
         final long taskExecutionNanos = timedRunnable.getTotalExecutionNanos();
-        assert taskExecutionNanos >= 0 : "expected task to always take longer than 0 nanoseconds, got: " + taskExecutionNanos;
-        executionEWMA.addValue(taskExecutionNanos);
+        assert taskExecutionNanos >= 0 || taskExecutionNanos == -1 :
```

Comment:
I am a bit paranoid about things. I'd prefer if we also tracked if an exception occurred with a boolean that gets updated in the `onRejection` and `onFailure` methods. Then we can say for certain that a taskExecutionNanos of -1 was really due to an exception.